src/components/gutter-control/gutter-wrapper.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { cloneElement, isValidElement } from '@wordpress/element';
+
+/**
+ * Return an element Wrapped with Gutter Properties.
+ *
+ * @return {JSX.Element}  GutterWrapper component
+ */
+function GutterWrapper( { children, gutter, gutterCustom, className, condition = true } ) {
+	if ( ! isValidElement( children ) ) {
+		return children;
+	}
+
+	let style = {};
+
+	if ( children.props.style ) {
+		style = {
+			...children.props.style,
+		};
+	}
+
+	if ( 'custom' === gutter ) {
+		style = {
+			...style,
+			'--coblocks-custom-gutter': `${ gutterCustom }em`,
```

Comment:
Block validation is failing when a custom gutter has been applied. You can see this by adding a block which supports gutters, setting a custom gutter value, saving and reloading the screen.

The style prop is always rendering the default value when it expects the custom value. This is the error message:

```
Content generated by `save` function:

<div class="wp-block-coblocks-services"><div class="has-columns has-2-columns has-responsive-columns has-custom-gutter" style="--coblocks-custom-gutter:1.6em"></div></div>

Content retrieved from post body:

<div class="wp-block-coblocks-services"><div class="has-columns has-2-columns has-responsive-columns has-custom-gutter" style="--coblocks-custom-gutter:5em">

</div></div>
```

The block validation process takes the block html in the post_content and runs the detectected attributes through the save function. If both the retrieved output and the save funciton's output are the same the block will pass validation.

Reply:
@jrtashjian turns out I was saying the attribute is a string (because eventually I want it to store a number and the unit) and I was saving a number because that's the output of <RangeControl/>. All fixed in latest commit.